semgrep-core/Parsing/other/yaml_to_generic.ml

```diff
@@ -172,15 +176,135 @@ let parse env parser : A.expr =
   in
   read_stream ()
 
+let make_pattern_expr e =
+  match e with
+  | A.Container(A.Dict, (_lp, [x], _rp)) -> A.E x
```

Comment:
comment here: "if a pattern contains just a field and a value, the user probably want to only
match this field, not the whole enclosing container".
I guess we don't need the PartialSingleField trick. Cool.